src/function/scalar/math/numeric.cpp

```diff
@@ -206,6 +206,21 @@ struct BitCntOperator {
 	}
 };
 
+struct HugeIntBitCntOperator {
+	template <class TA, class TR>
+	static inline TR Operation(TA input) {
+		using TU = typename std::make_unsigned<int64_t>::type;
+		TR count = 0;
+		for (auto value = TU(input.upper); value > 0; value >>= 1) {
```

Comment:
You might want to use Kernighan's algorithm for this:

```cpp
for (auto value = TU(input.upper); value; ++count) {
	value &= (value - 1);
}
```

Reply:
Done - thanks for the suggestion.